### PR TITLE
Extensions

### LINKED ISSUE #2: How to use this module for *.ts files?
Hi,
this module looks promising. Atm I use my custom script for cached transpilation, but I want to replace that with your module.

The problem is, I have a typescript project and want to make `babel-changed` to transpile my .ts files but they are ignored.

My current bash
```bash
babel-changed --src src --dest dist/lib --file-glob \"**/*.ts\"
```

Output:

```
> babel-changed --src src --dest dist/lib --file-glob "**/*.ts"

[babel] copying 76 files
[babel] nothing to compile
babel-changed: 56.8ms
```

From the output it looks like the ts-files are only copied, not transpiled.

EDIT:

Ok so it looks like this is intentional. https://github.com/smartprix/babel-changed/blob/master/index.js#L44
Babel has the [config option](https://babeljs.io/docs/en/babel-preset-typescript) `--extensions ".ts"` so maybe we should also add this option to babel-changed.

### PR DESCRIPTION
**Pull request by rohit-gohri**
Closes #2 

- Add extensions support
- Upgrade dependencies
- Update readme

### CODE REVIEW

**Comment by pubkey** [comment]
Wow. Thank you very much.

File: index.js [hunk 1]
```diff
@@ -41,7 +43,7 @@ async function transform({
 		const destFile = `${destDir}${srcFile.substring(srcDir.length)}`;
 		const mtimeDest = await file(destFile).mtime();
 		if (mtimeDest < mtimeSrc) {
-			if (srcFile.endsWith('.js')) {
+			if (extensions.reduce((result, extension) => result ||srcFile.endsWith(extension), false)) {
```
Comment (hit12369) [inline index.js]: `Array.some` can be used here instead.

**Review by hit12369 — CHANGES_REQUESTED** [review]

**Comment by hit12369** [comment]
Great work :+1:

File: bin/babel-changed.js [hunk 2]
```diff
@@ -18,6 +19,7 @@ const options = {
 	destDir: program.dest,
 	filesGlobPattern: program.fileGlob,
 	ignoredGlobPattern: program.ignoreGlob,
+	extensions: program.extensions.split(',').filter(Boolean),
```
Comment (hit12369) [inline bin/babel-changed.js]: Extensions should be trimmed, in case someone uses `-e "js, mjs, ts"`

**Review by hit12369 — APPROVED** [review]

**Pull request merged**